tests/test_stdiomgr_base.py

```diff
@@ -71,6 +68,26 @@ def test_context_manager_instance():
     assert list(cm) == value_list
 
 
+def test_context_manager_mro():
+    """Confirm StdioManager instance has correct MRO."""
+    cm = StdioManager()
+
+    mro = cm.__class__.__mro__
+
+    assert mro == (
```

Comment:
I hope this will help understanding.  It could be annotated with comments of the methods inherited along the way.

And I think there should be similar for each new class that has any significant inheritance complexity.

Reply:
Yeah, and we could set the docs up with individual, class-specific inheritance diagrams for each one ... \*sigh\*, and, ideally there would be a way to automatically generate a diagram with the method inheritance cascade in the docs, so that we don't have to curate it manually.